### PR TITLE
fix!: verify payload signatures

### PR DESCRIPTION
**Pull request by chingor13**
* For all requests, verify the signature before taking any actions.
* When queuing a task, re-sign the request body (which is not necessarily the webhook payload)

Not really a breaking change, but forcing a major version bump.

### CODE REVIEW

File: packages/gcf-utils/src/gcf-utils.ts [hunk 1]
```diff
@@ -57,14 +57,14 @@ interface Scheduled {
...
   body: string;
-  signature: string;
   id: string;
   name: string;
 }
 
 export interface WrapOptions {
   background: boolean;
   logging: boolean;
+  skipVerification?: boolean;
```
Comment (SurferJeffAtGoogle) [inline packages/gcf-utils/src/gcf-utils.ts:67]: Please add a comment for `skipVerification`.  Comments for `background` and `logging` would also be appreciated.

File: packages/gcf-utils/src/gcf-utils.ts [hunk 2]
```diff
@@ -232,6 +232,18 @@ export class GCFBootstrapper {
     }
   }
 
+  private static parseSignatureHeader(request: express.Request): string {
```
Comment (SurferJeffAtGoogle) [inline packages/gcf-utils/src/gcf-utils.ts:235]: In a comment for this function, explain why it returns 'unset'.  I would have expected it to return the empty string when a signature was not found in the request.

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]

File: packages/gcf-utils/test/gcf-bootstrapper.ts [hunk 3]
```diff
@@ -640,6 +640,115 @@ describe('GCFBootstrapper', () => {
...
+        });
+        req.body = {
+          installation: {id: 1},
+        };
+        req.headers = {};
+        req.headers['x-github-event'] = 'issues';
+        req.headers['x-github-delivery'] = '123';
+        req.headers['x-cloudtasks-taskname'] = 'my-task';
+        req.headers['x-hub-signature'] =
+          'sha1=c012c260559a04cf285e05d67e1ecedcad71b931';
```
Comment (tmatsuo) [inline packages/gcf-utils/test/gcf-bootstrapper.ts:710]: I'll be glad if there's a comment describing how to calculate this value.

**Review by tmatsuo — APPROVED** [review]
Looks good
Please address comments from @SurferJeffAtGoogle tho

File: packages/gcf-utils/src/gcf-utils.ts (hunk 1 above)
Comment (chingor13) [reply packages/gcf-utils/src/gcf-utils.ts:75]: Added comments for each

File: packages/gcf-utils/src/gcf-utils.ts (hunk 2 above)
Comment (chingor13) [reply packages/gcf-utils/src/gcf-utils.ts:251]: Done

File: packages/gcf-utils/test/gcf-bootstrapper.ts (hunk 3 above)
Comment (chingor13) [reply packages/gcf-utils/test/gcf-bootstrapper.ts:710]: Added a comment: `echo -n '{"installation":{"id":1}}' | openssl dgst -sha1 -hmac "foo"`

**Review by tmatsuo — APPROVED** [review]

**Review by SurferJeffAtGoogle — APPROVED** [review]

**Pull request merged**